app/uk/gov/hmrc/servicedependencies/persistence/BobbyRulesSummaryRepo.scala

```diff
@@ -16,78 +16,73 @@
+  def clearAllData: Future[Boolean]
 }
 
 @Singleton
-class BobbyRulesSummaryRepoImpl @Inject()(mongo: ReactiveMongoComponent)
-  extends ReactiveRepository[BobbyRulesSummary, BSONObjectID](
-      collectionName = "bobbyRulesSummary"
-    , mongo          = mongo.mongoConnector.db
-    , domainFormat   = BobbyRulesSummary.mongoFormat
+class BobbyRulesSummaryRepository @Inject()(mongo: MongoComponent)(implicit ec: ExecutionContext)
+    extends PlayMongoCollection[BobbyRulesSummary](
+      collectionName = "bobbyRulesSummary",
+      mongoComponent = mongo,
+      domainFormat   = BobbyRulesSummary.mongoFormat,
+      indexes = Seq(
+        IndexModel(ascending("date"), IndexOptions().name("dateIdx").unique(true))
+      )
     )
-  with BobbyRulesSummaryRepo {
-
-    private implicit val brsf = BobbyRulesSummary.mongoFormat
-    implicit val ldf = LocalDateFormats.localDateFormat
-    import ExecutionContext.Implicits.global
-
-    override def indexes: Seq[Index] =
-      Seq(Index(
-        Seq("date" -> IndexType.Ascending),
-        name   = Some("dateIdx"),
-        unique = true))
-
-    def add(summary: BobbyRulesSummary): Future[Unit] =
-      collection
-        .update(
-            selector = Json.obj("date" -> Json.toJson(summary.date))
-          , update   = summary
-          , upsert   = true
-          )
-        .map(_ => ())
-
-    def getLatest: Future[Option[BobbyRulesSummary]] =
-      find("date" -> Json.toJson(LocalDate.now))
-        .map(_.headOption)
-        .flatMap {
-          case Some(a) => Future(Some(a))
-          case None => getHistoric.map(_.headOption)
-        }
-
-    // Not time bound yet
-    def getHistoric: Future[List[BobbyRulesSummary]] =
-      collection.find(Json.obj())
-        .sort(Json.obj("date" -> -1))
-        .cursor[BobbyRulesSummary]()
-        .collect(maxDocs = -1, Cursor.FailOnError[List[BobbyRulesSummary]]())
-
-    def clearAllData: Future[Boolean] =
-      super.removeAll().map(_.ok)
-
-}
\ No newline at end of file
+    with BobbyRulesSummaryRepo {
+
+  private implicit val brsf = BobbyRulesSummary.mongoFormat
+  implicit val ldf          = LocalDateFormats.localDateFormat
+
+  def add(summary: BobbyRulesSummary): Future[Unit] =
+    collection
+      .replaceOne(
+        filter = equal("date", summary.date),
+        summary,
+        ReplaceOptions().upsert(true)
+      )
+      .toFuture()
+      .map(_ => ())
+
+  def getLatest: Future[Option[BobbyRulesSummary]] =
+    collection.find(equal("date", LocalDate.now)).toFuture()
+      .map(_.headOption)
+      .flatMap {
+        case Some(a) => Future(Some(a))
+        case None    => getHistoric.map(_.headOption)
+      }
+
+  // Not time bound yet
+  def getHistoric: Future[List[BobbyRulesSummary]] =
+    collection
+      .find()
+      .sort(descending("date")).toFuture().map(_.toList)
+
+  def clearAllData: Future[Boolean] =
+    collection.deleteMany(new BasicDBObject()).toFuture.map(_.wasAcknowledged())
```

Comment:
is it worth changing this to `dropCollection`? If not, I'll change the dropCollections I've defined to this.

Reply:
There is an admin controller which ends up hitting this, so dropping would have slightly different semantics and remove the indices and the underlying collection as well.